app/src/main/java/org/simple/clinic/editpatient/EditPatientViewRenderer.kt

```diff
@@ -19,5 +20,15 @@ class EditPatientViewRenderer(private val ui: EditPatientUi) : ViewRenderer<Edit
     }
 
     ui.setDateOfBirthAndAgeVisibility(fieldToShow)
+
+    manageButtonState(model)
+  }
+
+  private fun manageButtonState(model: EditPatientModel) {
+    if (model.saveButtonState == ButtonState.SAVING) {
+      ui.showProgressUntilPatientUpdated()
+    } else {
+      ui.patientUpdated()
```

Comment:
This state is not exactly `patientUpdated` but more like `patient update is not happening`. Can we rename the methods to better reflect this?

In addition, the UI need not care about the conditions for showing the progress since that responsibility is a part of the UI renderer and we can drop the `UntilPatientUpdated` from the method name.

Suggestions:
- `showProgress`
- `hideProgress`